packages/google_maps_flutter/google_maps_flutter_platform_interface/lib/src/types/bitmap.dart

```diff
@@ -954,3 +986,93 @@ class BytesMapBitmap extends MapBitmap {
         }
       ];
 }
+
+/// Represents a [BitmapDescriptor] that is created from a pin configuration.
+/// Can only be used with [AdvancedMarker]s.
+///
+/// The [backgroundColor] and [borderColor] are used to define the color of the
+/// standard pin marker.
+///
+/// The [glyph] parameter is used to define the glyph that is displayed on the
+/// pin marker (default glyph is a circle).
+///
+/// The following example demonstrates how to change colors of the default map
+/// pin to white and blue:
+///
+/// ```dart
+/// PinConfig(
+///   backgroundColor: Colors.blue,
+///   borderColor: Colors.white,
+///   glyph: Glyph.color(Colors.blue)
+/// )
+/// ```
+///
+/// The following example demonstrates how to customize a marker pin by showing
+/// a short text on the pin:
+///
+/// ```dart
+/// PinConfig(
+///   backgroundColor: Colors.blue,
+///   glyph: Glyph.text('Pin', Colors.white)
+/// )
+/// ```
+///
+/// The following example demonstrates how to customize a marker pin by showing
+/// a custom image on the pin:
+///
+/// ```dart
+/// PinConfig(
+///   glyph: Glyph.bitmapDescriptor(
+///     BitmapDescriptor.asset(
+///       ImageConfiguration(size: Size(12, 12)),
+///       'assets/cat.png'
+///    )
+/// )
+/// ```
+///
+class PinConfig extends BitmapDescriptor {
```

Comment:
>What's the reason for making this a BitmapDescription, given that it's not a bitmap? It feels out of place here.

I tried making it consistent with [Android implementation](https://developers.google.com/android/reference/com/google/android/gms/maps/model/BitmapDescriptorFactory#fromPinConfig(com.google.android.gms.maps.model.PinConfig)):

```java
AdvancedMarkerOptions advancedMarkerOptions =
    new AdvancedMarkerOptions()
        .icon(BitmapDescriptorFactory.fromPinConfig(pinConfig))
        .position(MARKER_POSITION);
```

>And it seems very problematic that PinConfig is a BitmapDescriptor, which has a Glyph as a field, and Glyph can be constructed with Glyph.bitmap(BitmapDescriptor bitmapDescriptor), which could be passed a PinConfig, allowing for arbitrary nesting of BitmapDescriptors in ways that make no sense.

I've added an assert to `BitmapGlyph` about `bitmap` not being a `AdvancedMarkerGlyph` to prevent nesting of unnecessary pin configs.